### DISCUSSION TITLE
Arduino UART bootloader

### DISCUSSION THREAD

**Discussion by daveythacher** [earlephilhower/arduino-pico#452] [Ideas]
How hard would it be to create a UART bootloader for the Arduino over FTDI 232R? Maybe use the on-board USB?

This would likely need a loader. The Arduino is modified Harvard and does not support execution from RAM. The RP2040 does and in many ways relies on it.

Modifying the bootloader logic to use UART should be possible. This would be booted by the on-board logic using the required stage2. Installing this bootloader would work using the built-in bootloader which does not require any programmers. From this point on the loader would work just like a standard UART loader. Note this will require flash to work.

The challenge here would be reworking the SDK to support this. However, if the UART logic consumes the hex file directly this may not be required.

**Comment by sequoia-hope** [comment]
Looks like this was added in August, after your comment.
https://github.com/earlephilhower/arduino-pico/blob/master/libraries/PicoOTA/examples/OTAfromFile/OTAfromFile.ino
